relay-server/src/services/processor.rs

```diff
@@ -504,6 +504,31 @@ impl<'a> ProcessEnvelopeState<'a> {
         self.remove_event();
         self.managed_envelope.reject_event(outcome);
     }
+
+    /// Determines whether profiles that would otherwise be dropped by dynamic sampling should be kept.
+    fn forward_unsampled_profiles(&self, global_config: &GlobalConfig) -> bool {
+        let Some(global_options) = &global_config.options else {
+            return false;
+        };
+
+        if !global_options.unsampled_profiles_enabled {
+            return false;
+        }
+
+        let event_platform = self
+            .event
+            .value()
+            .and_then(|e| e.platform.as_str())
+            .unwrap_or("");
+
+        self.project_state
+            .has_feature(Feature::IngestUnsampledProfiles)
+            && global_options
+                .profile_metrics_allowed_platforms
+                .iter()
+                .any(|s| s == event_platform)
+            && rand::random::<f32>() < global_options.profile_metrics_sample_rate
```

Comment:
Should use a `rand::thread_rng` here

Reply:
`random()` is a [convenience wrapper](https://docs.rs/rand/latest/rand/fn.random.html) around `thread_rng()`. Do you mean we should create a dedicated RNG here?

Reply:
oh TIL thanks 👍